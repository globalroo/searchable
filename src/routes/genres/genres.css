.genres-wrapper {
    border-radius: 5px;
    font-family: 'Roboto', Arial, Helvetica, sans-serif !important;
    margin-left: auto;
    margin-right: auto;
    margin-top: 40px;
    max-width: 940px;
    padding-left: 20px;
    padding-right: 20px;
}

.spacer {
    padding-bottom: 20px;
}