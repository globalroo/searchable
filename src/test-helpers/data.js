import { MEDIA_TYPE } from "src/tmdb-service/tmdb-api";

export const mockMovieItem = {
	id: 0,
	title: "movieItem",
	media_type: MEDIA_TYPE.MOVIE
};

export const mockTvItem = {
	id: 1,
	title: "tvItem",
	media_type: MEDIA_TYPE.TV
};

export const mockPersonItem = {
	id: 2,
	name: "personItem",
	media_type: MEDIA_TYPE.PERSON
};

export const mockResultsArray = [mockMovieItem, mockTvItem, mockPersonItem];

export const fakePeopleResponse = {
	page: 1,
	results: [
		{
			adult: false,
			gender: 2,
			name: "Tom Cruise",
			id: 500,
			known_for: [
				{
					id: 137113,
					video: false,
					vote_count: 7716,
					vote_average: 7.6,
					title: "Edge of Tomorrow",
					release_date: "2014-05-27",
					original_language: "en",
					original_title: "Edge of Tomorrow",
					genre_ids: [28, 878],
					backdrop_path: "/7mgKeg18Qml5nJQa56RBZO7dIu0.jpg",
					adult: false,
					overview:
						"Major Bill Cage is an officer who has never seen a day of combat when he is unceremoniously demoted and dropped into combat. Cage is killed within minutes, managing to take an alpha alien down with him. He awakens back at the beginning of the same day and is forced to fight and die again... and again - as physical contact with the alien has thrown him into a time loop.",
					poster_path: "/tpoVEYvm6qcXueZrQYJNRLXL88s.jpg",
					popularity: 19.518,
					media_type: "movie"
				},
				{
					adult: false,
					backdrop_path: "/8TE77jL2e4zdERpv8hnBAHUmFRx.jpg",
					genre_ids: [28, 878, 12, 9648],
					id: 75612,
					original_language: "en",
					original_title: "Oblivion",
					overview:
						"Jack Harper is one of the last few drone repairmen stationed on Earth.  Part of a massive operation to extract vital resources after decades of war with a terrifying threat known as the Scavs, Jackâ€™s mission is nearly complete.  His existence is brought crashing down when he rescues a beautiful  stranger from a downed spacecraft.  Her arrival triggers a chain of events that  forces him to question everything he knows and puts the fate of humanity in his hands.",
					poster_path: "/hmOzkHlkGvi8x24fYpFSnXvjklv.jpg",
					release_date: "2013-04-10",
					title: "Oblivion",
					video: false,
					vote_average: 6.5,
					vote_count: 6645,
					media_type: "movie"
				},
				{
					adult: false,
					backdrop_path: "/pc7a2qrIkIxzqWGqcexd3mHzIxy.jpg",
					genre_ids: [28, 53, 12],
					id: 56292,
					original_language: "en",
					original_title: "Mission: Impossible - Ghost Protocol",
					overview:
						"Ethan Hunt and his team are racing against time to track down a dangerous terrorist named Hendricks, who has gained access to Russian nuclear launch codes and is planning a strike on the United States. An attempt to stop him ends in an explosion causing severe destruction to the Kremlin and the IMF to be implicated in the bombing, forcing the President to disavow them. No longer being aided by the government, Ethan and his team chase Hendricks around the globe, although they might still be too late to stop a disaster.",
					poster_path: "/s58mMsgIVOFfoXPtwPWJ3hDYpXf.jpg",
					release_date: "2011-12-07",
					title: "Mission: Impossible - Ghost Protocol",
					video: false,
					vote_average: 6.9,
					vote_count: 5905,
					popularity: 17.303,
					media_type: "movie"
				}
			],
			known_for_department: "Acting",
			profile_path: "/2Dkx4uuGoWFrPSitxdikv9z5azR.jpg",
			popularity: 14.983
		},
		{
			adult: false,
			gender: 2,
			name: "Quentin Tarantino",
			id: 138,
			known_for: [
				{
					adult: false,
					backdrop_path: "/qUcmEqnzIwlwZxSyTf3WliSfAjJ.jpg",
					genre_ids: [18, 37],
					id: 68718,
					original_language: "en",
					original_title: "Django Unchained",
					overview:
						"With the help of a German bounty hunter, a freed slave sets out to rescue his wife from a brutal Mississippi plantation owner.",
					poster_path: "/5WJnxuw41sddupf8cwOxYftuvJG.jpg",
					release_date: "2012-12-25",
					title: "Django Unchained",
					video: false,
					vote_average: 8.0,
					vote_count: 15120,
					popularity: 29.141,
					media_type: "movie"
				},
				{
					adult: false,
					backdrop_path: "/4cDFJr4HnXN5AdPw4AKrmLlMWdO.jpg",
					genre_ids: [53, 80],
					id: 680,
					original_language: "en",
					original_title: "Pulp Fiction",
					overview:
						"A burger-loving hit man, his philosophical partner, a drug-addled gangster's moll and a washed-up boxer converge in this sprawling, comedic crime caper. Their adventures unfurl in three stories that ingeniously trip back and forth in time.",
					poster_path: "/dM2w364MScsjFf8pfMbaWUcWrR.jpg",
					release_date: "1994-09-10",
					title: "Pulp Fiction",
					video: false,
					vote_average: 8.4,
					vote_count: 14543,
					popularity: 25.172,
					media_type: "movie"
				},
				{
					id: 16869,
					video: false,
					vote_count: 11377,
					vote_average: 8.1,
					title: "Inglourious Basterds",
					release_date: "2009-08-18",
					original_language: "en",
					original_title: "Inglourious Basterds",
					genre_ids: [28, 18, 53, 10752],
					backdrop_path: "/7nF6B9yCEq1ZCT82sGJVtNxOcl5.jpg",
					adult: false,
					overview:
						'In Nazi-occupied France during World War II, a group of Jewish-American soldiers known as "The Basterds" are chosen specifically to spread fear throughout the Third Reich by scalping and brutally killing Nazis. The Basterds, lead by Lt. Aldo Raine soon cross paths with a French-Jewish teenage girl who runs a movie theater in Paris which is targeted by the soldiers.',
					poster_path: "/ai0LXkzVM3hMjDhvFdKMUemoBe.jpg",
					popularity: 22.04,
					media_type: "movie"
				}
			],
			known_for_department: "Directing",
			profile_path: "/rivHScdhT043n5jCSci7ToqLsoE.jpg",
			popularity: 9.14
		},
		{
			adult: false,
			gender: 2,
			name: "Brad Pitt",
			id: 287,
			known_for: [
				{
					adult: false,
					backdrop_path: "/52AfXWuXCHn3UjD17rBruA9f5qb.jpg",
					genre_ids: [18],
					id: 550,
					original_language: "en",
					original_title: "Fight Club",
					overview:
						'A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy. Their concept catches on, with underground "fight clubs" forming in every town, until an eccentric gets in the way and ignites an out-of-control spiral toward oblivion.',
					poster_path: "/adw6Lq9FiC9zjYEpOqfq03ituwp.jpg",
					release_date: "1999-10-15",
					title: "Fight Club",
					video: false,
					vote_average: 8.4,
					vote_count: 15667,
					popularity: 29.185,
					media_type: "movie"
				},
				{
					id: 16869,
					video: false,
					vote_count: 11385,
					vote_average: 8.1,
					title: "Inglourious Basterds",
					release_date: "2009-08-18",
					original_language: "en",
					original_title: "Inglourious Basterds",
					genre_ids: [28, 18, 53, 10752],
					backdrop_path: "/7nF6B9yCEq1ZCT82sGJVtNxOcl5.jpg",
					adult: false,
					overview:
						'In Nazi-occupied France during World War II, a group of Jewish-American soldiers known as "The Basterds" are chosen specifically to spread fear throughout the Third Reich by scalping and brutally killing Nazis. The Basterds, lead by Lt. Aldo Raine soon cross paths with a French-Jewish teenage girl who runs a movie theater in Paris which is targeted by the soldiers.',
					poster_path: "/ai0LXkzVM3hMjDhvFdKMUemoBe.jpg",
					popularity: 22.76,
					media_type: "movie"
				},
				{
					adult: false,
					backdrop_path: "/A0WKbRIojF9DUWG4XLCmg5JK6I5.jpg",
					genre_ids: [80, 9648, 53],
					id: 807,
					original_language: "en",
					original_title: "Se7en",
					overview:
						'Two homicide detectives are on a desperate hunt for a serial killer whose crimes are based on the "seven deadly sins" in this dark and haunting film that takes viewers from the tortured remains of one victim to the next. The seasoned Det. Sommerset researches each sin in an effort to get inside the killer\'s mind, while his novice partner, Mills, scoffs at his efforts to unravel the case.',
					poster_path: "/8zw8IL4zEPjkh8Aysdcd0FwGMb0.jpg",
					release_date: "1995-09-22",
					title: "Se7en",
					video: false,
					vote_average: 8.3,
					vote_count: 10142,
					popularity: 29.462,
					media_type: "movie"
				}
			],
			known_for_department: "Acting",
			profile_path: "/kU3B75TyRiCgE270EyZnHjfivoq.jpg",
			popularity: 16.385
		},
		{
			adult: false,
			gender: 2,
			name: "Leonardo DiCaprio",
			id: 6193,
			known_for: [
				{
					adult: false,
					backdrop_path: "/s2bT29y0ngXxxu2IA8AOzzXTRhd.jpg",
					genre_ids: [28, 878, 12],
					id: 27205,
					original_language: "en",
					original_title: "Inception",
					overview:
						"Cobb, a skilled thief who commits corporate espionage by infiltrating the subconscious of his targets is offered a chance to regain his old life as payment for a task considered to be impossible: \"inception\", the implantation of another person's idea into a target's subconscious.",
					poster_path: "/qmDpIHrmpJINaRKAfWQfftjCdyi.jpg",
					release_date: "2010-07-15",
					title: "Inception",
					video: false,
					vote_average: 8.3,
					vote_count: 21372,
					popularity: 34.216,
					media_type: "movie"
				},
				{
					adult: false,
					backdrop_path: "/qUcmEqnzIwlwZxSyTf3WliSfAjJ.jpg",
					genre_ids: [18, 37],
					id: 68718,
					original_language: "en",
					original_title: "Django Unchained",
					overview:
						"With the help of a German bounty hunter, a freed slave sets out to rescue his wife from a brutal Mississippi plantation owner.",
					poster_path: "/5WJnxuw41sddupf8cwOxYftuvJG.jpg",
					release_date: "2012-12-25",
					title: "Django Unchained",
					video: false,
					vote_average: 8.0,
					vote_count: 15123,
					popularity: 20.027,
					media_type: "movie"
				},
				{
					adult: false,
					backdrop_path: "/vFUI5obFtx4IdhP6k8Om5ezHTrk.jpg",
					genre_ids: [18, 10749, 53],
					id: 597,
					original_language: "en",
					original_title: "Titanic",
					overview:
						"101-year-old Rose DeWitt Bukater tells the story of her life aboard the Titanic, 84 years later. On April 10, 1912, a young Rose boards the ship with her mother and fiancÃ©. Meanwhile, Jack Dawson and Fabrizio De Rossi win third-classic tickets. Rose tells the whole story from departure through the death of Titanic on its first and last voyage, on April 15, 1912.",
					poster_path: "/kHXEpyfl6zqn8a6YuozZUujufXf.jpg",
					release_date: "1997-11-18",
					title: "Titanic",
					video: false,
					vote_average: 7.8,
					vote_count: 13594,
					popularity: 23.946,
					media_type: "movie"
				}
			],
			known_for_department: "Acting",
			profile_path: "/jqbqNrOIB3alGMX6Gh2MbOKMXZO.jpg",
			popularity: 14.851
		},
		{
			adult: false,
			gender: 2,
			name: "Jackie Chan",
			id: 18897,
			known_for: [
				{
					adult: false,
					backdrop_path: "/pmLEmMLYCreFAxfdCt5uO1wDgSi.jpg",
					genre_ids: [12, 16, 10751, 35],
					id: 9502,
					original_language: "en",
					original_title: "Kung Fu Panda",
					overview:
						"When the Valley of Peace is threatened, lazy Po the panda discovers his destiny as the \"chosen one\" and trains to become a kung fu hero, but transforming the unsleek slacker into a brave warrior won't be easy. It's up to Master Shifu and the Furious Five -- Tigress, Crane, Mantis, Viper and Monkey -- to give it a try.",
					poster_path: "/2Paj1nufT0jeSY0G4u3RC31HIGT.jpg",
					release_date: "2008-06-04",
					title: "Kung Fu Panda",
					video: false,
					vote_average: 7.0,
					vote_count: 5814,
					popularity: 16.216,
					media_type: "movie"
				},
				{
					adult: false,
					backdrop_path: "/eyiE1vuizMKVu8MoZkg2xsramFl.jpg",
					genre_ids: [16, 10751],
					id: 49444,
					original_language: "en",
					original_title: "Kung Fu Panda 2",
					overview:
						"Po is now living his dream as The Dragon Warrior, protecting the Valley of Peace alongside his friends and fellow kung fu masters, The Furious Five - Tigress, Crane, Mantis, Viper and Monkey. But Poâ€™s new life of awesomeness is threatened by the emergence of a formidable villain, who plans to use a secret, unstoppable weapon to conquer China and destroy kung fu. It is up to Po and The Furious Five to journey across China to face this threat and vanquish it. But how can Po stop a weapon that can stop kung fu? He must look to his past and uncover the secrets of his mysterious origins; only then will he be able to unlock the strength he needs to succeed.",
					poster_path: "/Fzyy2dauoKxjUzUPYYo3kJkJAW.jpg",
					release_date: "2011-05-25",
					title: "Kung Fu Panda 2",
					video: false,
					vote_average: 6.7,
					vote_count: 3447,
					popularity: 14.504,
					media_type: "movie"
				},
				{
					adult: false,
					backdrop_path: "/rfhnEYubWD4hnUUm8xXmotpqffb.jpg",
					genre_ids: [28, 12, 18, 10751],
					id: 38575,
					original_language: "en",
					original_title: "The Karate Kid",
					overview:
						"12-year-old Dre Parker could have been the most popular kid in Detroit, but his mother's latest career move has landed him in China. Dre immediately falls for his classmate Mei Ying but the cultural differences make such a friendship impossible. Even worse, Dre's feelings make him an enemy of the class bully, Cheng. With no friends in a strange land, Dre has nowhere to turn but maintenance man Mr. Han, who is a kung fu master. As Han teaches Dre that kung fu is not about punches and parries, but maturity and calm, Dre realizes that facing down the bullies will be the fight of his life.",
					poster_path: "/bucxUHJd1DFy15iHzuXSOoFrObW.jpg",
					release_date: "2010-06-10",
					title: "The Karate Kid",
					video: false,
					vote_average: 6.3,
					vote_count: 2996,
					media_type: "movie"
				}
			],
			known_for_department: "Acting",
			profile_path: "/pmKJ4sGvPQ3imzXaFnjW4Vk5Gyc.jpg",
			popularity: 18.136
		},
		{
			adult: false,
			gender: 2,
			name: "Bruce Willis",
			id: 62,
			known_for: [
				{
					adult: false,
					backdrop_path: "/4cDFJr4HnXN5AdPw4AKrmLlMWdO.jpg",
					genre_ids: [53, 80],
					id: 680,
					original_language: "en",
					original_title: "Pulp Fiction",
					overview:
						"A burger-loving hit man, his philosophical partner, a drug-addled gangster's moll and a washed-up boxer converge in this sprawling, comedic crime caper. Their adventures unfurl in three stories that ingeniously trip back and forth in time.",
					poster_path: "/dM2w364MScsjFf8pfMbaWUcWrR.jpg",
					release_date: "1994-09-10",
					title: "Pulp Fiction",
					video: false,
					vote_average: 8.4,
					vote_count: 14564,
					popularity: 30.195,
					media_type: "movie"
				},
				{
					adult: false,
					backdrop_path: "/9pkZesKMnblFfKxEhQx45YQ2kIe.jpg",
					genre_ids: [27, 53],
					id: 381288,
					original_language: "en",
					original_title: "Split",
					overview:
						"Though Kevin has evidenced 23 personalities to his trusted psychiatrist, Dr. Fletcher, there remains one still submerged who is set to materialize and dominate all the others. Compelled to abduct three teenage girls led by the willful, observant Casey, Kevin reaches a war for survival among all of those contained within him â€” as well as everyone around him â€” as the walls between his compartments shatter apart.",
					poster_path: "/rXMWOZiCt6eMX22jWuTOSdQ98bY.jpg",
					release_date: "2017-01-19",
					title: "Split",
					video: false,
					vote_average: 7.2,
					vote_count: 9777,
					popularity: 29.333,
					media_type: "movie"
				},
				{
					adult: false,
					backdrop_path: "/vuHG8VlYnvw4gUKtS7GY5kZol69.jpg",
					genre_ids: [28, 53, 878],
					id: 59967,
					original_language: "en",
					original_title: "Looper",
					overview:
						"In the futuristic action thriller Looper, time travel will be invented but it will be illegal and only available on the black market. When the mob wants to get rid of someone, they will send their target 30 years into the past where a looper, a hired gun, like Joe is waiting to mop up. Joe is getting rich and life is good until the day the mob decides to close the loop, sending back Joe's future self for assassination.",
					poster_path: "/sNjL6SqErDBE8OUZlrDLkexfsCj.jpg",
					release_date: "2012-09-26",
					title: "Looper",
					video: false,
					vote_average: 6.8,
					vote_count: 6530,
					popularity: 13.524,
					media_type: "movie"
				}
			],
			known_for_department: "Acting",
			profile_path: "/zoTPQXh3qLd0Nj8JCVZqdWowQw7.jpg",
			popularity: 12.242
		},
		{
			adult: false,
			gender: 1,
			name: "Scarlett Johansson",
			id: 1245,
			known_for: [
				{
					id: 24428,
					video: false,
					vote_count: 18514,
					vote_average: 7.6,
					title: "The Avengers",
					release_date: "2012-04-25",
					original_language: "en",
					original_title: "The Avengers",
					genre_ids: [28, 12, 878],
					backdrop_path: "/hbn46fQaRmlpBuUrEiFqv0GDL6Y.jpg",
					adult: false,
					overview:
						"When an unexpected enemy emerges and threatens global safety and security, Nick Fury, director of the international peacekeeping agency known as S.H.I.E.L.D., finds himself in need of a team to pull the world back from the brink of disaster. Spanning the globe, a daring recruitment effort begins!",
					poster_path: "/cezWGskPY5x7GaglTTRN4Fugfb8.jpg",
					popularity: 66.207,
					media_type: "movie"
				},
				{
					id: 271110,
					video: false,
					vote_count: 13162,
					vote_average: 7.3,
					title: "Captain America: Civil War",
					release_date: "2016-04-27",
					original_language: "en",
					original_title: "Captain America: Civil War",
					genre_ids: [28, 12, 878],
					backdrop_path: "/m5O3SZvQ6EgD5XXXLPIP1wLppeW.jpg",
					adult: false,
					overview:
						"Following the events of Age of Ultron, the collective governments of the world pass an act designed to regulate all superhuman activity. This polarizes opinion amongst the Avengers, causing two factions to side with Iron Man or Captain America, which causes an epic battle between former allies.",
					poster_path: "/kSBXou5Ac7vEqKd97wotJumyJvU.jpg",
					popularity: 38.732,
					media_type: "movie"
				},
				{
					id: 99861,
					video: false,
					vote_count: 12488,
					vote_average: 7.3,
					title: "Avengers: Age of Ultron",
					release_date: "2015-04-22",
					original_language: "en",
					original_title: "Avengers: Age of Ultron",
					genre_ids: [28, 12, 878],
					backdrop_path: "/rFtsE7Lhlc2jRWF7SRAU0fvrveQ.jpg",
					adult: false,
					overview:
						"When Tony Stark tries to jumpstart a dormant peacekeeping program, things go awry and Earthâ€™s Mightiest Heroes are put to the ultimate test as the fate of the planet hangs in the balance. As the villainous Ultron emerges, it is up to The Avengers to stop him from enacting his terrible plans, and soon uneasy alliances and unexpected action pave the way for an epic and unique global adventure.",
					poster_path: "/t90Y3G8UGQp0f0DrP60wRu9gfrH.jpg",
					popularity: 55.953,
					media_type: "movie"
				}
			],
			known_for_department: "Acting",
			profile_path: "/tHMgW7Pg0Fg6HmB8Kh8Ixk6yxZw.jpg",
			popularity: 23.809
		}
	],
	total_pages: 1000,
	total_results: 20000
};
