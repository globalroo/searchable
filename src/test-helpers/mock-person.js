export const mockPerson = {
	birthday: "1962-07-03",
	known_for_department: "Acting",
	deathday: null,
	id: 500,
	name: "Tom Cruise",
	images: {
		profiles: [
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 5,
				height: 1500,
				vote_average: 5.0350140056022,
				file_path: "/nPGebm4wxlLThpEEjTKthjoHMbM.jpg",
				width: 1000
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 6,
				height: 1500,
				vote_average: 5.1138716356108,
				file_path: "/hIVsySa8Ro6vrQ0hygIF6ylBgK7.jpg",
				width: 1000
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 8,
				height: 1500,
				vote_average: 5.1173708920188,
				file_path: "/fttvtGtgjPjoKEqGtnebbwEkVNA.jpg",
				width: 1000
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 22,
				height: 2100,
				vote_average: 5.282,
				file_path: "/3oWEuo0e8Nx8JvkqYCDec2iMY6K.jpg",
				width: 1400
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 7,
				height: 1050,
				vote_average: 5.1972789115646,
				file_path: "/hOoTazc5VZdXvMTLGpu9H7nb1e7.jpg",
				width: 700
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 11,
				height: 1800,
				vote_average: 5.302,
				file_path: "/54wOvUh3vWOTDqPMDq1VTYuh9yv.jpg",
				width: 1200
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 8,
				height: 1800,
				vote_average: 5.392,
				file_path: "/2Dkx4uuGoWFrPSitxdikv9z5azR.jpg",
				width: 1200
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 0,
				height: 600,
				vote_average: 0,
				file_path: "/s57wicK5nPuYhglUFUWECjnomJL.jpg",
				width: 400
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 0,
				height: 840,
				vote_average: 0,
				file_path: "/9pTIJwjH3Wg8JAxt0TNTpu0cXK0.jpg",
				width: 560
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 3,
				height: 600,
				vote_average: 5.318,
				file_path: "/gThaIXgpCm3PCiXwFNDBJCme85y.jpg",
				width: 400
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 0,
				height: 1500,
				vote_average: 0,
				file_path: "/bn9CwMSM3o4XPZEPZdQ3grslGup.jpg",
				width: 1000
			}
		]
	},
	also_known_as: [
		"Том Круз",
		"トム・クルーズ",
		"ทอม ครูซ",
		"湯姆·克魯斯",
		"톰 크루즈",
		"توم كروز",
		"Thomas Cruise Mapother IV",
		"Τομ Κρουζ",
		"טום קרוז"
	],
	gender: 2,
	biography:
		'Thomas "Tom" Cruise (born Thomas Cruise Mapother IV; July 3, 1962) is an American actor and filmmaker. He has been nominated for three Academy Awards and has won three Golden Globe Awards.\n\nHe started his career at age 19 in the 1981 film Endless Love. After portraying supporting roles in Taps (1981) and The Outsiders (1983), his first leading role was in Risky Business, released in August 1983. Cruise became a full-fledged movie star after starring as Pete "Maverick" Mitchell in Top Gun (1986). He has since 1996 been well known for his role as secret agent Ethan Hunt in the Mission: Impossible film series. One of the biggest movie stars in Hollywood, Cruise has starred in many successful films, including The Color of Money (1986), Cocktail (1988), Rain Man (1988), Born on the Fourth of July (1989), Far and Away(1992), A Few Good Men (1992), The Firm (1993), Interview with the Vampire: The Vampire Chronicles (1994), Jerry Maguire (1996), Eyes Wide Shut (1999), Magnolia (1999), Vanilla Sky (2001), Minority Report (2002),The Last Samurai (2003), Collateral (2004), War of the Worlds (2005), Lions for Lambs (2007), Valkyrie (2008), Knight and Day (2010), Jack Reacher (2012), Oblivion (2013), and Edge of Tomorrow (2014).\n\nIn 2012, Cruise was Hollywood\'s highest-paid actor. Fifteen of his films grossed over $100 million domestically; twenty-one have grossed in excess of $200 million worldwide. Cruise is known for his support for the Church of Scientology and its affiliated social programs.',
	popularity: 14.983,
	place_of_birth: "Syracuse, New York, USA",
	profile_path: "/2Dkx4uuGoWFrPSitxdikv9z5azR.jpg",
	adult: false,
	imdb_id: "nm0000129",
	homepage: "http://www.tomcruise.com/"
};

export const mockCast = {
	cast: [
		{
			id: 74,
			character: "Ray Ferrier",
			original_title: "War of the Worlds",
			overview:
				"Ray Ferrier is a divorced dockworker and less-than-perfect father. Soon after his ex-wife and her new husband drop of his teenage son and young daughter for a rare weekend visit, a strange and powerful lightning storm touches down.",
			vote_count: 4208,
			video: false,
			media_type: "movie",
			poster_path: "/xXMM9KY2eq1SDOQif9zO91YOBA8.jpg",
			backdrop_path: "/bwTb66gicrB7HbkogwNs9V37Ddb.jpg",
			popularity: 16.077,
			title: "War of the Worlds",
			original_language: "en",
			genre_ids: [12, 878, 53],
			vote_average: 6.3,
			adult: false,
			release_date: "2005-06-29",
			credit_id: "52fe4213c3a36847f800226b"
		}
	]
};

export const mockCrew = {
	crew: [
		{
			id: 616,
			department: "Production",
			original_language: "en",
			original_title: "The Last Samurai",
			job: "Producer",
			overview:
				"Nathan Algren is an American hired to instruct the Japanese army in the ways of modern warfare, which finds him learning to respect the samurai and the honorable principles that rule them. Pressed to destroy the samurai's way of life in the name of modernization and open trade, Algren decides to become an ultimate warrior himself and to fight for their right to exist.",
			vote_count: 3267,
			video: false,
			media_type: "movie",
			release_date: "2003-12-05",
			vote_average: 7.4,
			title: "The Last Samurai",
			popularity: 15.985,
			genre_ids: [18, 28, 10752, 36],
			backdrop_path: "/lm15lu2jt3I3Syxsb90UJzUzMRV.jpg",
			adult: false,
			poster_path: "/sLv5pXysIz7QbtKFJy85d5yxv2W.jpg",
			credit_id: "52fe425ec3a36847f8018e1f"
		}
	]
};
