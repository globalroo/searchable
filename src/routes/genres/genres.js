import React, { useEffect, useState } from "react";

import { CategoryLoader } from "src/components/category-loader";
import { AssetScroller } from "src/components/asset-scroller";

import { NavigateBackContainer } from "src/components/navigate-back";
import { getDiscoverMoviesByGenreId } from "src/tmdb-service/tmdb-api";
import { getMovieGenres } from "src/tmdb-service/tmdb-api";

import "./genres.css";

export const GenreCategory = ({ id, name }) => {
	return (
		<CategoryLoader
			fetcher={getDiscoverMoviesByGenreId}
			title={name}
			id={id}
			data-testid={`genre-category-${id}`}
		>
			{results => (
				<div className="spacer">
					<AssetScroller assets={results} />
				</div>
			)}
		</CategoryLoader>
	);
};

export const PopularMoviesByGenre = () => {
	const [genres, setGenres] = useState([]);

	useEffect(() => {
		getMovieGenres().then(({ payload }) => {
			const { genres } = payload;
			console.log(genres);
			setGenres(genres);
		});
	}, []);

	if (genres.length === 0) return <div>Loading...</div>;

	return (
		<div className="genres-wrapper">
			<NavigateBackContainer>
				<h1 className="bold-header-uppercase">Genres</h1>
			</NavigateBackContainer>
			<p>Most popular movies in each Genre</p>
			{genres.map(({ name, id }) => (
				<GenreCategory key={id} name={name} id={id} />
			))}
		</div>
	);
};
