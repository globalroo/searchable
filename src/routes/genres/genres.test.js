import React from "react";
import { render, cleanup, waitForElement } from "react-testing-library";
import { PopularMoviesByGenre } from "./genres";
import "jest-dom/extend-expect";
import mockConsole from "jest-mock-console";

import { mockMovieItem } from "src/test-helpers/data";

jest.mock("src/tmdb-service/tmdb-api", () => {
	const tmdbAPI = jest.requireActual("src/tmdb-service/tmdb-api");
	return {
		MEDIA_TYPE: tmdbAPI.MEDIA_TYPE,
		getMovieGenres: () =>
			Promise.resolve({
				payload: {
					genres: [{ name: "test-genre1", id: 1 }, { name: "test-genre2", id: 2 }]
				}
			}),
		getDiscoverMoviesByGenreId: () =>
			Promise.resolve({ response: { ok: true }, payload: { results: [mockMovieItem] } })
	};
});

const restoreConsole = mockConsole();

describe("Trending component Renders as expected", () => {
	afterAll(restoreConsole);
	afterEach(cleanup);
	it("Renders a component with a navigate back wrapper", async () => {
		const { container, getByTestId } = render(<PopularMoviesByGenre />);
		await waitForElement(() => getByTestId("genre-category-1"));
		expect(container).toMatchSnapshot();
		expect(container).toHaveTextContent("test-genre2");
		expect(container).toHaveTextContent("test-genre1");
	});
});
