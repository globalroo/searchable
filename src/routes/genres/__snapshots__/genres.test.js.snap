// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trending component Renders as expected Renders a component with a navigate back wrapper 1`] = `
<div>
  <div
    class="genres-wrapper"
  >
    <div
      class="pointer-hover"
      style="display: flex; flex-direction: row; align-items: flex-start; justify-content: flex-start;"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-1"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
      <h1
        class="bold-header-uppercase"
      >
        Genres
      </h1>
    </div>
    <p>
      Most popular movies in each Genre
    </p>
    <div
      data-testid="genre-category-1"
    >
      <h1>
        test-genre1
      </h1>
      <div
        class="spacer"
      >
        <div
          style="display: flex; flex-wrap: wrap; justify-content: space-between; overflow: hidden;"
        >
          <ul
            class="MuiGridList-root-10"
            data-testid="asset-grid"
            style="margin: -2px; flex-wrap: nowrap; transform: translateZ(0); padding-top: 3px; padding-bottom: 3px;"
          >
            <div
              class="MuiGridListTile-root-11 pointer-hover image-tile"
              data-testid="0_movieItem"
              style="padding: 5px; height: 225px; width: 150px;"
            >
              <div
                class="MuiGridListTile-tile-12"
              >
                <img
                  alt="movieItem"
                  src="http://image.tmdb.org/t/p/w300/undefined"
                />
                <div
                  class="MuiGridListTileBar-root-15 MuiGridListTileBar-titlePositionBottom-16"
                >
                  <div
                    class="MuiGridListTileBar-titleWrap-19"
                  >
                    <div
                      class="MuiGridListTileBar-title-22"
                    >
                      movieItem
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </ul>
        </div>
      </div>
    </div>
    <div
      data-testid="genre-category-2"
    >
      <h1>
        test-genre2
      </h1>
      <div
        class="spacer"
      >
        <div
          style="display: flex; flex-wrap: wrap; justify-content: space-between; overflow: hidden;"
        >
          <ul
            class="MuiGridList-root-10"
            data-testid="asset-grid"
            style="margin: -2px; flex-wrap: nowrap; transform: translateZ(0); padding-top: 3px; padding-bottom: 3px;"
          >
            <div
              class="MuiGridListTile-root-11 pointer-hover image-tile"
              data-testid="0_movieItem"
              style="padding: 5px; height: 225px; width: 150px;"
            >
              <div
                class="MuiGridListTile-tile-12"
              >
                <img
                  alt="movieItem"
                  src="http://image.tmdb.org/t/p/w300/undefined"
                />
                <div
                  class="MuiGridListTileBar-root-15 MuiGridListTileBar-titlePositionBottom-16"
                >
                  <div
                    class="MuiGridListTileBar-titleWrap-19"
                  >
                    <div
                      class="MuiGridListTileBar-title-22"
                    >
                      movieItem
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
